### PR TITLE
Fix gcc sanitizer runtime errors

### PR DESCRIPTION
**Pull request by suburbski**
Fixes (common) modsided errors showing up at runtime with the following instrumentation flags:
`-fsanitize=address -fsanitize-address-use-after-scope -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=leak -fsanitize=undefined`

### CODE REVIEW

**Comment by suburbski** [comment]
Also added `-fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow`

File: src/game/q_shared.h [hunk 1]
```diff
@@ -1520,8 +1520,8 @@ typedef enum
 
 typedef struct entityState_s
 {
-	int number;                     // entity index
-	entityType_t eType;             // entityType_t
+	int number;             // entity index
+	int eType;              // changed from entityType_t to int to allow ET_EVENTS + eventNum
```
Comment (Aciz) [inline src/game/q_shared.h:1524]: I don't suppose this causes any problems @isRyven ? You probably know better.

**Review by Aciz — COMMENTED** [review]
LGTM but could you check this too @isRyven in case I missed something?

File: src/cgame/cg_playerstate.cpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ void CG_CheckAmmo(void)
 
 	for (i = 0 ; i < WP_NUM_WEAPONS ; i++)
 	{
-		if (!(weapons[0] & (1 << i)))
+		if (!(static_cast<long>(weapons[0]) & (1L << i)))
```
Comment (suburbski) [inline src/cgame/cg_playerstate.cpp:35]: This prevents the overflow here but does the cast make any sense when `weapons[0]` is an int when the flags are set?

**Comment by isRyven** [comment]
is this ready?

**Comment by suburbski** [comment]
yes

**Comment by isRyven** [comment]
alright, thank you

**Pull request merged**